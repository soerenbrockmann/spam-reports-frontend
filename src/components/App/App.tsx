import React, { Component } from 'react';
import { ReportList } from '../../containers/ReportList';

export class App extends Component {
    render() {
        return <ReportList />;
    }
}
