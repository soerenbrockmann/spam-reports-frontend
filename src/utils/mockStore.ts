import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const mockStore = configureStore([thunk]);

export const store = mockStore({
    reports: { size: 25, nextOffset: '', elements: [] },
});
