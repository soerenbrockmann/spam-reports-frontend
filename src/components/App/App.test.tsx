import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { App } from './App';
import { store } from '../../utils/mockStore';

describe('<App>', () => {
    it.only('renders correctly', () => {
        const component = renderer.create(
            <Provider store={store}>
                <App />
            </Provider>
        );
        expect(component).toMatchSnapshot();
    });
});
