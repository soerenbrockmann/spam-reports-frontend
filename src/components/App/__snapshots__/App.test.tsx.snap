// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> renders correctly 1`] = `
<div>
  LOADING
  <h1>
    Reports (
    25
    )
  </h1>
</div>
`;
